packages/core/src/common/config-manager.ts

```diff
@@ -0,0 +1,182 @@
+import ts from 'typescript';
+import { fileURLToPath } from 'node:url';
+import { dirname, extname, resolve } from 'node:path';
+import type { Options as PrettierOptions } from 'prettier';
+import { createRequire } from 'node:module';
+
+export type TsUserConfigLang = 'typescript' | 'javascript';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+const require = createRequire(import.meta.url);
+
+export interface TSUserConfig {
+  preferences?: TsUserPreferencesConfig;
+  suggest?: TSSuggestConfig;
+  format?: TsFormatConfig;
+}
+
+export interface TsUserPreferencesConfig {
+  importModuleSpecifier: ts.UserPreferences['importModuleSpecifierPreference'];
```

Comment:
Are these keys all specific to how the TS extension for Code stores its preferences? If so, would it be cleaner to avoid baking those keys into `@glint/core` and instead have it use the standard interfaces (like `ts.UserPreferences`/`ts.EditorSettings`/etc), leaving it the responsibility of `glint-vscode` to translate Code-specific config to the appropriate format?